pkg/stanza/operator/input/windows/input.go

```diff
@@ -216,6 +216,14 @@ func (i *Input) read(ctx context.Context) int {
 	events, err := i.subscription.Read(i.maxReads)
 	if err != nil {
 		i.Logger().Error("Failed to read events from subscription", zap.Error(err))
+		i.Logger().Info("Closing subscription")
+		closeErr := i.subscription.Close()
+		if closeErr != nil {
+			i.Logger().Error("Failed to close subscription", zap.Error(closeErr))
+			return 0
+		}
+		i.Logger().Info("Resubscribing")
+		i.subscription = NewRemoteSubscription(i.remote.Server)
```

Comment:
I think we should only create a new subscription when we receive `handle is invalid`

Reply:
Agreed that resubscribe should require a more specific error. We will also need to resubscribe on `subscription handle is not open` as `handle is invalid` only happens once if we close the subscription and then the collector starts logging `subscription handle is not open` when we re-create before the remote host is back up.